### DISCUSSION TITLE
Phone Auth

### DISCUSSION THREAD

**Discussion by AntonioLi1** [invertase/react-native-firebase#6697] [Q&A]
Hi everyone!
Are users able to register an account with phone authentication?
I noticed for email and password auth, it explicitly mentions it can be used for register AND login. But in the phone auth page, it doesn't mention register.
Does anyone know if phone auth can register new users?
If so, does it work differently to logging in existing users?

Thank you!!

**Comment by mikehardy** [comment]
https://firebase.google.com/docs/auth/web/phone-auth#next-steps

> After a user signs in for the first time, a new user account is created and linked to the credentials—that is, the user name and password, phone number, or auth provider information—the user signed in with. This new account is stored as part of your Firebase project, and can be used to identify a user across every app in your project, regardless of how the user signs in.

So how did it go when you tried it? Just signing in should have created the user entry, did it not work?

That said, keep this in mind:

https://firebase.google.com/docs/auth/web/phone-auth#security-concerns

> Authentication using only a phone number, while convenient, is less secure than the other available methods, because possession of a phone number can be easily transferred between users. Also, on devices with multiple user profiles, any user that can receive SMS messages can sign in to an account using the device's phone number.
> 
> If you use phone number based sign-in in your app, you should offer it alongside more secure sign-in methods, and inform users of the security tradeoffs of using phone number sign-in.

**Reply by AntonioLi1** [reply]
When i tried signup, it took me to the emulator browser and did a not "verify youre not a robot" check, then returned to the app and proceeded successfully.
But will the direction to the browser always happen on a real phone (iOS and android)?

In addition, is it possible to send a OTP to the emulator android instead of testing it via the documented way of creating a phone number and OTP on firebase?

**Reply by mikehardy** [reply]
> But will the direction to the browser always happen on a real phone (iOS and android)?

How did it go when you tested it on a real phone? (on a real phone if you have setup APNS correctly it should not need the captcha, but it might if you have not set up APNS correctly, for me it does not, as mine is setup correctly)

> In addition, is it possible to send a OTP to the emulator android instead of testing it via the documented way of creating a phone number and OTP on firebase?

Emulators don't have phone numbers that can receive OTPs via SMS, so I do not know how that would work.

**Reply by AntonioLi1** [reply]
Im unable to test on a real phone just yet. What do you mean by APNS?

Cool thank you!

**Reply by mikehardy** [reply]
> Im unable to test on a real phone just yet.

That's not something that is viable then. You cannot expect to implement phone auth without a real phone

> APNS

https://search.brave.com/search?q=Apple+APNS
